cdc/owner.go

```diff
@@ -120,25 +123,35 @@ func (c *changeFeed) reAddTable(id, startTs uint64) {
 	}
 }
 
-func (c *changeFeed) addTable(id, startTs uint64, table schema.TableName) {
+func (c *changeFeed) addTable(sid, tid, startTs uint64, table schema.TableName) {
 	if c.filter.ShouldIgnoreTable(table.Schema, table.Table) {
 		return
 	}
 
-	c.tables[id] = table
-	c.orphanTables[id] = model.ProcessTableInfo{
-		ID:      id,
+	if _, ok := c.schemas[sid]; !ok {
+		c.schemas[sid] = make(tableIDMap)
+	}
+	c.schemas[sid][tid] = struct{}{}
+	c.tables[tid] = table
+	c.orphanTables[tid] = model.ProcessTableInfo{
+		ID:      tid,
 		StartTs: startTs,
 	}
 }
 
-func (c *changeFeed) removeTable(id uint64) {
-	delete(c.tables, id)
+func (c *changeFeed) removeTable(sid, tid uint64) {
+	if _, ok := c.schemas[sid]; ok {
+		delete(c.schemas[sid], tid)
+		if len(c.schemas[sid]) == 0 {
+			delete(c.schemas, sid)
```

Comment:
It's OK to have an empty database with no table. It might be simpler if we only do `c.schemas` cleanup when a database is dropped.